tests/test_spawner.py

```diff
@@ -125,6 +127,64 @@ async def test_spawn(kube_ns, kube_client, config):
     assert isinstance(status, int)
 
 
+@pytest.mark.asyncio
+async def test_spawn_pending_pods(kube_ns, kube_client):
+    """Spawner can deal with pods that are pending."""
+    c = Config()
+    # Make a config where the fake-jupyter pod cannot be scheduled so it
+    # will always be in a pending state.
+    # You can use any setting here that will be impossible to satisfy on the
+    # minikube cluster. 
+    c.KubeSpawner.node_selector = {'disktype': 'ssd'}
+    c.KubeSpawner.namespace = kube_ns
+    c.KubeSpawner.start_timeout = 30  # Do not wait very long.
```

Comment:
```suggestion
    c.KubeSpawner.start_timeout = 5  # Do not wait very long.
```

This duration will increase the time to complete the tests significantly, so I suggest to make this duration quite low.